cpp/src/utils/metrics/registry.hh

```diff
@@ -0,0 +1,24 @@
+#ifndef TS_CPP_UTILS_METRICS_REGISTRY_HH_
+#define TS_CPP_UTILS_METRICS_REGISTRY_HH_
+
+#include <stdexcept>
+
+#include "src/utils/metrics/log_metrics_cache.hh"
+#include "src/utils/metrics/yaml_config.hh"
+
+namespace torchserve {
+class MetricsRegistry {
+ public:
+  static void Initialize(const std::string& metrics_config_file_path,
+                         const MetricsContext& metrics_context);
+  static const std::shared_ptr<MetricsConfigurationHandler>&
+  GetMetricsConfigurationHandlerInstance();
+  static std::shared_ptr<MetricsCache>& GetMetricsCacheInstance();
+
+ private:
+  static std::shared_ptr<MetricsConfigurationHandler> metrics_config_handler;
```

Comment:
it is not necessary to waste space to store this variable which is only used for testing. 

yaml parser can have its test case to cover this part.